Added lab_1 template, added tests for tokenize function

dmitry-uraev opened the pull request.

dmitry-uraev committed:
Added lab_1 template, added tests fot tokenize function

dmitry-uraev committed:
Added tests for text preprocessing functions

dmitry-uraev committed:
Added tests for some of the work-with-dictionary functions

dmitry-uraev committed:
Added basic tests for work-with-dictionary functions; advanced function TBD

dmitry-uraev committed:
Refactored some outputs

WhiteJaeger commented on lab_1/dictionary_test.py:
```diff
@@ -0,0 +1,214 @@
+        self.assertEqual(get_concordance(['happy', 'man'], 'happy', -1, 0), ['happy'])
+        self.assertEqual(get_concordance(['happy', 'man'], 'happy', 0, 1), ['happy man'])
+        self.assertEqual(get_concordance(['happy', 'man'], 'man', -1, 0), ['man'])
+        bad_inputs = [[], {}, 'string', (), None, 9.34, True]
+        for bad_input in bad_inputs:
+            self.assertEqual(get_concordance(['happy', 'man', 'went'], 'man', bad_input, bad_input), [])
```
Should not it be in another test?
(The next one is identical to it)

dmitry-uraev replied:
yes, that is true, I ll change it

WhiteJaeger commented on lab_1/dictionary_test.py:
```diff
@@ -0,0 +1,214 @@
+        """
+        Checks that function can handle incorrect inputs
+        """
+        bad_inputs = [{}, 'string', (), None, 9, 9.34, True]
+        for bad_input in bad_inputs:
+            self.assertEqual(get_concordance(bad_input, 'happy', 2, 3), [])
+        bad_inputs = [{}, [], (), None, 9, 9.34, True]
```
May be concatenate them in one so there is no code duplication and unnecessary assigning?

```suggestion
        bad_inputs = [{}, [], 'string', (), None, 9, 9.34, True]
        for bad_input in bad_inputs:
            self.assertEqual(get_concordance(bad_input, 'happy', 2, 3), [])
```

dmitry-uraev replied:
yep, it is possible to avoid duplication, but `bad_inputs` variable changes a little due to argument testing, e.g `'string'` is not allowed for the first argument, but allowed for the second. I'll refactor this test so, please, check it again then.

WhiteJaeger commented on lab_1/dictionary_test.py:
```diff
@@ -0,0 +1,214 @@
+        """
+        Checks that function can handle incorrect inputs
+        """
+        bad_inputs = [{}, 'string', (), None, 9, 9.34, True]
+        for bad_input in bad_inputs:
+            self.assertEqual(get_adjacent_words(bad_input, 'happy', 2, 3), [])
+        bad_inputs = [{}, [], (), None, 9, 9.34, True]
```
Same as in the previous comment.

WhiteJaeger commented on lab_1/main.py:
```diff
@@ -0,0 +1,125 @@
+    """
+    Splits sentences into tokens, converts the tokens into lowercase, removes punctuation
+    :param text: the initial text
+    :return: a list of lowercase tokens without punctuation
```
Typo: `lowercased`

dmitry-uraev replied:
PyCharm gives typo in "lowercased" :) `Typo: In word 'lowercased'` Will be corrected

WhiteJaeger commented:
General comment for tests:
It is better visually and easier to read if we split `expected` and `actual` results in different variables.

Example
```
expected = ['1', '2', '3']
actual = function_name('1 2 3')
self.assertEqual(expected, actual)
```

dmitry-uraev commented:
> General comment for tests:
> It is better visually and easier to read if we split `expected` and `actual` results in different variables.
> 
> Example
> 
> ```
> expected = ['1', '2', '3']
> actual = function_name('1 2 3')
> self.assertEqual(expected, actual)
> ```

TBD by the next meeting

dmitry-uraev committed:
Restyled all tests, concatenated some tests, removed duplicates in code

dmitry-uraev committed:
Added one more test to check right context sort

dmitry-uraev committed:
Merge remote-tracking branch 'upstream/master'

dmitry-uraev committed:
Setting files for plagiarism check

WhiteJaeger committed:
Skip labs unittests

WhiteJaeger committed:
Skip labs

WhiteJaeger commented:
I have skipped unittests for now. 
Now it is necessary to deal with lint & plagiarism.

WhiteJaeger committed:
Add reqs for plagiarism check

WhiteJaeger committed:
Add skipping for templates

WhiteJaeger commented:
I have dealt with lint.
The last thing to consider is plagiarism check: even if it is green now, i do not think it is working correctly.

Further note:
Create separate branch for feature, do no develop in master as it is easier to maintain several features and more convenient for other people who use your fork.

dmitry-uraev commented:
> I have skipped unittests for now.
> Now it is necessary to deal with lint & plagiarism.

![image](https://user-images.githubusercontent.com/43416920/92565900-1b8c3600-f284-11ea-95d4-424a9fe46ce3.png)

we have to somehow add public GitHub token for accessing all pull requests and then everything should work properly it seems

WhiteJaeger commented:
> > I have skipped unittests for now.
> > Now it is necessary to deal with lint & plagiarism.
> 
> ![image](https://user-images.githubusercontent.com/43416920/92565900-1b8c3600-f284-11ea-95d4-424a9fe46ce3.png)
> 
> we have to somehow add public GitHub token for accessing all pull requests and then everything should work properly it seems

I was be able to access raw files without token. I will try to investigate furthermore.

dmitry-uraev committed:
added lab_1 description

WhiteJaeger commented:
I suggest we split this PR into two:

1. The one with lab template (leave it here);
2. The other one with plagiarism check as it may take some time to implement.

dmitry-uraev committed:
Removed plagiarism checking files

demid5111 commented on lab_1/lab_work_1.md:
```diff
@@ -0,0 +1,154 @@
+Конкорданс – это список контекстов искомого слова в исследуемом корпусе (тексте).
+Данные списки полезны для изучения устойчивых словосочетаний (коллокаций).
+
+Подробную информацию про конкорданс вы можете найти в данных источниках:
```
need example in description.

demid5111 commented on lab_1/lab_work_1.md:
```diff
@@ -0,0 +1,154 @@
+Данные списки полезны для изучения устойчивых словосочетаний (коллокаций).
+
+Подробную информацию про конкорданс вы можете найти в данных источниках:
+* http://www.lomonosov-fund.ru/enc/ru/encyclopedia:0127200:article
```
additional links to the end

demid5111 commented on lab_1/lab_work_1.md:
```diff
@@ -0,0 +1,154 @@
+* https://core.ac.uk/download/pdf/83100161.pdf
+
+Для выполнения  лабораторной работы достаточно понимать,
+что конкорданс - это список примеров употребления определенного слова в заданном тексте.
```
yo

demid5111 commented on lab_1/lab_work_1.md:
````diff
@@ -0,0 +1,154 @@
+  pass
+```
+
+### Шаг 5. Получить конкорданс по заданному слову. Выполнение Шагов 1-5 соответствует 7 баллам
````
7?? move to 6

demid5111 commented on lab_1/lab_work_1.md:
```diff
@@ -0,0 +1,154 @@
+Например,
+'Mary was quick to realize that she had won the prize...'.
+* Слово – 'realize'. Ограничение на количество слов в левом контексте – 3, в правом – 1.
+* Функция возвращает один пример (потенциально их может быть несколько) – 'was quick to realize that'.
```
returns a list (!)

demid5111 replied:
list of lists?

demid5111 commented on lab_1/lab_work_1.md:
````diff
@@ -0,0 +1,154 @@
+
+Интерфейс: 
+```py
+def get_concordance(tokens: list, word: str, left_range: int, right_range: int) -> list:
````
left_context_size, right_context_size

demid5111 replied:
testing right_range? left_range?

demid5111 commented on lab_1/lab_work_1.md:
```diff
@@ -0,0 +1,154 @@
+'Mary was quick to realize that she had won the prize...'.
+* Слово – 'realize'. Ограничение на количество слов в левом контексте – 3, в правом – 1.
+* Функция возвращает один пример (потенциально их может быть несколько) – 'was quick to realize that'.
+В левом контексте 3 слова ('was quick to'), в правом 1 ('that').
```
where is requirement for not appropriate arguments?

demid5111 commented on lab_1/lab_work_1.md:
````diff
@@ -0,0 +1,154 @@
+
+Интерфейс: 
+```py
+def get_top_n_words(freq_dict: dict, top_n: int) -> list:
````
what is returned if argument is not OK

demid5111 commented on lab_1/lab_work_1.md:
````diff
@@ -0,0 +1,154 @@
+
+Интерфейс: 
+```py
+def read_from_file(path_to_file: str) -> str:
````
??

demid5111 commented on lab_1/lab_work_1.md:
````diff
@@ -0,0 +1,154 @@
+  pass
+```
+
+Запись результатов в файл. Название файла - `report.txt`. Лежит в корневой директории.
````
in root?

demid5111 commented on lab_1/lab_work_1.md:
````diff
@@ -0,0 +1,154 @@
+```
+
+Запись результатов в файл. Название файла - `report.txt`. Лежит в корневой директории.
+Внутренняя структура: каждая строка содержит пример употребления заданного слова.
````
concordance list is printed line by line

demid5111 commented on lab_1/lab_work_1.md:
````diff
@@ -0,0 +1,154 @@
+
+Интерфейс: 
+```py
+def write_to_file(path_to_file: str, content: tuple):
````
list

demid5111 commented on lab_1/lab_work_1.md:
````diff
@@ -0,0 +1,154 @@
+```
+
+
+### Шаг 7. Получить слово из левого или правого контекста. Выполнение Шагов 1-6 + 7 соответствует 9 баллам
````
or (!)

demid5111 commented on lab_1/lab_work_1.md:
```diff
@@ -0,0 +1,154 @@
+Например,
+'Mary was quick to realize that she had won the prize...'.
+* Слово – 'realize'. Нужно найти 3е слово в левом контексте, в правом – 1ое.
+* Функция возвращает одну пару слов (потенциально их может быть несколько) – [('was', 'that')]
```
list

demid5111 replied:
list brackets

demid5111 commented on lab_1/lab_work_1.md:
```diff
@@ -0,0 +1,154 @@
+* Слово – 'realize'. Нужно найти 3е слово в левом контексте, в правом – 1ое.
+* Функция возвращает одну пару слов (потенциально их может быть несколько) – [('was', 'that')]
+
+Интерфейс: 
```
requirement: re-use

demid5111 commented on lab_1/lab_work_1.md:
````diff
@@ -0,0 +1,154 @@
+
+Интерфейс: 
+```py
+def get_adjacent_words(tokens: list, word: str, left_n: int, right_n: int) -> list:
````
corner cases?

demid5111 commented on lab_1/lab_work_1.md:
```diff
@@ -0,0 +1,154 @@
+в которых данное слово встречается.
+
+Например,
+'Mary was quick to realize that she had won the prize...'.
```
[] instead of str

demid5111 commented on lab_1/lab_work_1.md:
````diff
@@ -0,0 +1,154 @@
+  pass
+```
+
+
````
exchange 6 with 7

demid5111 commented on lab_1/lab_work_1.md:
```diff
@@ -0,0 +1,154 @@
+Примеры сортируются по первому слову либо левого контекста, либо правого.
+
+Например,
+'Mary was quick to realize that she had won the prize. Then she realized she was disappointed'.
```
realize in both examples

demid5111 commented on lab_1/lab_work_1.md:
````diff
@@ -0,0 +1,154 @@
+
+Интерфейс: 
+```py
+def get_and_sort_concordance(tokens: list, word: str, left_range: int, right_range: int, left_sort=True) -> list:
````
no default values in function declaration

demid5111 commented on lab_1/main.py:
```diff
@@ -0,0 +1,125 @@
+    left_sort = True
+    --> ['dog is happy but the cat', 'man is happy the dog is']
+    """
+    pass
```
where is template?

Commit:
added start template

ValeriyaKuznetsova committed:
changed some issues in lab_work_1.md

dmitry-uraev committed:
added tests for numeric arguments covering upper horizon;
added [None] testing argument for bad inputs functions;
added new tests for 10 points function; test level splitting TBD

ValeriyaKuznetsova committed:
changed other issues in lab description

ValeriyaKuznetsova committed:
changed some issues in lab description

removed tuples, default values
changed return values from strings to lists of tokens

ValeriyaKuznetsova committed:
changed one minor issue in lab description

connected with "left_sort" incorrect input

dmitry-uraev committed:
Added data corpus and stop words corpus

dmitry-uraev committed:
Added test template for starter

dmitry-uraev committed:
fixed lint

dmitry-uraev committed:
Refactored some tests due to template changes

ValeriyaKuznetsova commented on lab_1/dictionary_test.py:
```diff
@@ -293,7 +293,7 @@ def test_get_and_left_sort_concordance_ideal(self):
         left_sort = True
 
         expected = ['dog is happy but the cat', 'man is happy the dog is']
```
now we expect tokens everywhere from get_concordance till here, like [['dog', 'is'], ['cat', 'is']]

ValeriyaKuznetsova commented on lab_1/dictionary_test.py:
```diff
@@ -0,0 +1,362 @@
+        """
+        Ideal get concordance scenario
+        """
+        expected = ['man is happy the dog is']
```
tokens here and later

ValeriyaKuznetsova commented on lab_1/dictionary_test.py:
```diff
@@ -0,0 +1,362 @@
+        self.assertEqual(expected, actual)
+        actual = get_concordance(['happy'], '', 2, 3)
+        self.assertEqual(expected, actual)
+        expected = ['happy']
```
nothing is expected here

ValeriyaKuznetsova commented on lab_1/dictionary_test.py:
```diff
@@ -0,0 +1,362 @@
+        """
+        Checks that function can handle incorrect number inputs
+        """
+        expected = ['happy']
```
nothing is expected here

ValeriyaKuznetsova commented on lab_1/dictionary_test.py:
```diff
@@ -0,0 +1,362 @@
+        expected = ['happy man']
+        actual = get_concordance(['happy', 'man'], 'happy', 0, 1)
+        self.assertEqual(expected, actual)
+        expected = ['man']
```
nothing is expected here

ValeriyaKuznetsova commented on lab_1/dictionary_test.py:
```diff
@@ -0,0 +1,362 @@
+        Checks if function can handle great right range numbers,
+        that exceed the number of given tokens
+        """
+        expected = ['happy man']
```
tokens

ValeriyaKuznetsova commented on lab_1/dictionary_test.py:
```diff
@@ -0,0 +1,362 @@
+        left_n = 2
+        right_n = 3
+
+        expected = [('man', 'is'), ('dog', 'cat')]
```
a list of lists is expected. No tuples anywhere

ValeriyaKuznetsova commented on lab_1/dictionary_test.py:
```diff
@@ -0,0 +1,362 @@
+        self.assertEqual(expected, actual)
+        actual = get_adjacent_words(['happy', 'man'], 'man', -1, 0)
+        self.assertEqual(expected, actual)
+        expected = [('', 'man')]
```
[['man']] is expected

ValeriyaKuznetsova commented on lab_1/dictionary_test.py:
```diff
@@ -0,0 +1,362 @@
+        tokens = ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy',
+                  'dog', 'is', 'happy', 'still', 'the', 'cat', 'is', 'sad']
+        word = 'happy'
+        left_range = 2
```
it is called "left_context_size" here and in get_concordance (same with 'right...')

dmitry-uraev committed:
Added testing lint implementation

dmitry-uraev committed:
Refactored some tests due to template change

ValeriyaKuznetsova commented on lab_1/dictionary_test.py:
```diff
@@ -0,0 +1,362 @@
+        Checks if function can handle great right range numbers,
+        that exceed the number of given tokens
+        """
+        expected = [['happy', 'man']]
```
only 'man' should be returned

ValeriyaKuznetsova commented on lab_1/dictionary_test.py:
```diff
@@ -0,0 +1,362 @@
+        Checks if function can handle great left range numbers,
+        that exceed the number of given tokens
+        """
+        expected = [['one', 'happy']]
```
only 'one' should be returned

ValeriyaKuznetsova commented on lab_1/dictionary_test.py:
```diff
@@ -0,0 +1,362 @@
+        tokens = ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy',
+                  'dog', 'is', 'happy', 'still', 'the', 'cat', 'is', 'sad']
+        word = 'happy'
+        left_range = 2
```
left_context_size and right_context_size

ValeriyaKuznetsova commented on lab_1/dictionary_test.py:
```diff
@@ -0,0 +1,362 @@
+        left_sort = False
+
+        expected = [['man', 'is', 'happy', 'dog', 'is', 'happy'], ['dog', 'is', 'happy', 'still', 'the', 'cat']]
+        actual = sort_concordance(tokens, word, left_range, right_range, left_sort)
```
left_context_size and right_context_size

dmitry-uraev committed:
Refactored some tests due to template change

WhiteJaeger committed:
Parse output

demid5111 commented on lab_1/lab_work_1.md:
```diff
@@ -0,0 +1,192 @@
+
+1. Создать форк репозитория
+2. Установить необходимые инструменты для работы
+3. Изменить файл `main.py`
```
start.py?

demid5111 commented on lab_1/lab_work_1.md:
```diff
@@ -0,0 +1,192 @@
+
+Если на вход подаются некорректные токены или слово, возвращается пустой список. 
+
+Если на вход подается одно некорректное значение из left_context_size или right_context_size, возвращается только правый или левый контекст вместе с исходным словом соответственно, например, [['the', 'weather', 'was', 'sunny'], ['today', 'it', 'is', 'sunny']] при исходном слове 'sunny', left_context_size = 3, right_context_size = 0.
```
what is incorrect?

demid5111 commented on lab_1/lab_work_1.md:
```diff
@@ -0,0 +1,192 @@
+
+### Шаг 6. Получить слово из левого и правого контекста. Выполнение Шагов 1-6 соответствует 8 баллам
+
+В данной функции ОБЯЗАТЕЛЬНО использовать функцию get_concordance (см. Шаг 5).
```
test (!)

demid5111 commented on lab_1/lab_work_1.md:
````diff
@@ -0,0 +1,192 @@
+
+Интерфейс: 
+```py
+def read_from_file(path_to_file: str) -> str:
````
start.py
  read_from_file()

main.py
  def read()

demid5111 replied:
read is implemented by us -> no ask to implement in description

demid5111 commented on lab_1/start.py:
```diff
@@ -0,0 +1,15 @@
+    # DO NOT REMOVE NEXT LINE - KEEP IT INTENTIONALLY LAST
+    assert RESULT == [(), ()], 'Concordance not working'
+
+# Test: a) assert concordance_realize is in file
```
to remove

demid5111 commented on lab_1/start.py:
```diff
@@ -0,0 +1,15 @@
+"""
+
+if __name__ == '__main__':
+    #  use data.txt and stop_words.txt files to test your program
```
read x2?

demid5111 replied:
# for example of such words you can open stop_words.txt
stop_words = []

WhiteJaeger committed:
Rm linting from pipeline

demid5111 commented on lab_1/start.py:
```diff
@@ -0,0 +1,15 @@
+
+    RESULT = None
+    # DO NOT REMOVE NEXT LINE - KEEP IT INTENTIONALLY LAST
+    assert RESULT == [(), ()], 'Concordance not working'
```
is

WhiteJaeger committed:
Fix conflicts

ValeriyaKuznetsova committed:
Update lab_work_1.md

ValeriyaKuznetsova committed:
Update lab_work_1.md

dmitry-uraev committed:
Added remarks

WhiteJaeger committed:
return reqs

ValeriyaKuznetsova committed:
Update lab_work_1.md

WhiteJaeger committed:
Add file reading

WhiteJaeger committed:
Return stop words
